kategory-docs/docs/docs/datatypes/ior/README.md

```diff
@@ -6,4 +6,125 @@ permalink: /docs/datatypes/ior/
+
+val both = Ior.Both("Warning", 3)
+```
+
+Kategory also offers extension functions for `Ior`, the `leftIor`, `rightIor` and `bothIor`:
+
+```kotlin:ank
+import kategory.*
+
+val right = 3.rightIor()
+
+val left = "Error".leftIor()
+
+val both = ("Warning" to 3).bothIor()
+```
+
+
+When we look at the `Monad` or `Applicative` instances of `Ior`, we can see that they actually require a `Semigroup` instance on the left side.
+This is because `Ior` will actually accumulate failures on the left side, very similarly to how the [`Validated`](/docs/datatypes/validated) data type does.
+This means we can accumulate data on the left side while also being able to short-circuit upon the first right-side-only value.
+For example, sometimes, we might want to accumulate warnings together with a valid result and only halt the computation on a "hard error"
+Here's an example of how we are able to do that:
+
+```kotlin:ank
+import kategory.*
+
+data class User(val name: String, val pw: String)
+
+fun validateUsername(username: String) = when {
+    username.isEmpty() -> Nel.of("Can't be empty").leftIor()
+    username.contains(".") -> (Nel.of("Dot in name is deprecated") to username).bothIor()
+    else -> username.rightIor()
+}
+
+
+fun validatePassword(password: String) = when {
+    password.length < 8 -> Nel.of("Password too short").leftIor()
+    password.length < 10 -> (Nel.of("Password should be longer") to password).bothIor()
+    else -> password.rightIor()
+}
+
+fun validateUser(name: String, pass: String) =
+        Ior.monad<Nel<String>>().binding {
+            val username = validateUsername(name).bind()
+            val password = validatePassword(pass).bind()
+            yields(User(username, password))
+        }.ev()
+
+```
+
+Now we're able to validate user data and also accumulate non-fatal warnings:
+
+```kotlin:ank
+
+validateUser("John", "password12")
+
+validateUser("john.doe", "password")
+
+validateUser("jane", "short")
+
+```
+
+To extract the values, we can use the `fold` method, which expects a function for each case the `Ior` can represent:
+
+```kotlin:ank
+
+validateUser("john.doe", "password").fold(
+        { "Error: ${it.head}" },
+        { "Success $it"},
+        { warnings, (name) -> "Warning: $name; The following warnings occurred: ${warnings.show()}" }
+)
+
+```
+Similar to [Validated](/docs/datatypes/validated), there is also a type alias for using a `NonEmptyList` on the left side.
+
+```kotlin
+typealias IorNel<A, B> = Ior<Nel<A>, B>
+```
+
+
+```kotlin:ank
+
+val left: IorNel<String, Int> = Ior.leftNel("Error")
+
+val both: IorNel<String, Int> = Ior.bothNel("Warning", 1)
+
+```
+
+
+We can also convert our `Ior` to `Either`, `Validated` or `Option`.
+All of these conversions will discard the left side value if both are available:
+
+```kotlin:ank
+Ior.Both("Warning", 42).toEither()
+```
+
+Available Instances:
+
+```kotlin:ank
+import kategory.debug.*
+
+showInstances<IorHK, Unit>()
+```
```

Comment:
This is missing proper credits stating that this tutorial is derived from https://typelevel.org/cats/datatypes/ior.html with links to it like other tutorials do.
We should try to not just port tutorial from cats but introduce original content where possible